util/helper_scripts_for_mongodb/update_gem5_versions.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+# Copyright (c) 2023 The Regents of the University of California
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met: redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer;
+# redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution;
+# neither the name of the copyright holders nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import argparse
+import json
+
+from bson import json_util
+from helper import (
+    get_database,
+    save_to_json,
+)
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "--uri", help="URI of the database", type=str, required=True
+)
+parser.add_argument(
+    "--version",
+    help="Version to add to gem5_versions",
+    type=str,
+    required=True,
+)
+parser.add_argument(
+    "--db_name", help="Name of the database", type=str, default="gem5-vision"
+)
+parser.add_argument(
+    "--collection_name",
+    help="Name of the collection",
+    type=str,
+    default="resources",
+)
+parser.add_argument(
+    "--category", help="Category to not update", action="append"
+)
+parser.add_argument(
+    "--outfile",
+    help="Name of the output json file",
+    type=str,
+    default="resources_update_gem5_versions.json",
+)
+
+args = parser.parse_args()
+
+uri = args.uri
+db_name = args.db_name
+collection_name = args.collection_name
+category_to_not_update = args.category
+version = args.version
+output_json_file = args.outfile
+
+collection = get_database(uri, db_name, collection_name)
+
+
+# get all documents from resources collection
+resources = collection.find({}, {"_id": 0})
```

Comment:
```suggestion
resources = collection.find({})
```

As far as I know `collection.find({})` will return all documents in the collection. I don't know what `{"_id":0}` does.

Reply:
The `{"_id":0}` just ignores that field and does not return it.

Reply:
Is there some reason we want to ignore this field? I would've thought we want (or need) to keep it. My reasoning is that when you update a resource entry  the database needs this key to know which entries to overwrite with new values. Otherwise how will it know the difference between one entry and another?

Reply:
We never use this field, so it does not matter if we ignore it or not. To update resources I use the resource id + resource version.